src/main/kotlin/org/cascadebot/bot/cmd/HelpCommand.kt

```diff
@@ -0,0 +1,324 @@
+package org.cascadebot.bot.cmd
+
+import net.dv8tion.jda.api.interactions.commands.Command
+import net.dv8tion.jda.api.interactions.commands.OptionType
+import net.dv8tion.jda.api.interactions.commands.build.CommandData
+import net.dv8tion.jda.api.interactions.commands.build.Commands
+import net.dv8tion.jda.api.interactions.commands.build.OptionData
+import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
+import org.cascadebot.bot.Main
+import org.cascadebot.bot.MessageType
+import org.cascadebot.bot.cmd.meta.CommandArgs
+import org.cascadebot.bot.cmd.meta.CommandContext
+import org.cascadebot.bot.cmd.meta.RootCommand
+import org.cascadebot.bot.db.entities.CommandOptionEntity
+import org.cascadebot.bot.db.entities.CustomCommandEntity
+import org.cascadebot.bot.db.entities.GuildSlotEntity
+import org.cascadebot.bot.utils.QueryUtils.queryEntity
+import org.cascadebot.bot.utils.tryOrNull
+import java.util.UUID
+
+class HelpCommand : RootCommand("help", "Give you a list of all commands available to be run") {
+    override val commandData: CommandData = Commands.slash(name, description)
+        .addOptions(OptionData(OptionType.STRING, "command", "Command to get help on", false))
+
+    override fun onCommand(context: CommandContext, args: CommandArgs) {
+        if (args.getAmount() == 0) {
+            handleNoArgs(context)
+        } else {
+            handleCommandHelp(context, args.getArgAsString("command")!!)
+        }
+    }
+
+    private fun handleNoArgs(context: CommandContext) {
+        // TODO spilt into 3 pages, root commands, custom commands, and modules, for now their all going to be on one page
+        val globalBuilder = StringBuilder()
+        context.jda.retrieveCommands().queue { commands ->
+            for (command in commands) {
+                val cascadeCommand = Main.commandManager.getCommand(command.name)
+                val slashCommandData = cascadeCommand?.commandData as SlashCommandData
+                globalBuilder.append("* </${command.name}:${command.id}> ")
+                for (option in slashCommandData.options) {
+                    if (option.isRequired) {
+                        globalBuilder.append("<${option.name}> ")
+                    } else {
+                        globalBuilder.append("[${option.name}] ")
+                    }
+                }
+                globalBuilder.append('\n')
+            }
+            val guildBuilder = StringBuilder()
+            context.guild.retrieveCommands().queue { guildCommands ->
+                for (slot in Main.postgres.transaction {
+                    queryEntity(GuildSlotEntity::class.java) { root ->
+                        equal(root.get<Long>("guildId"), context.guild.idLong)
+                    }.list()
+                }) {
```

Comment:
This could be replaced with a joined query to query the custom commands - this would remove the need to request each custom command individually